### ISSUE TITLE
Don't inject configurations using element templates, if user provides no value

### ISSUE THREAD

**Issue by MaxTru** [bpmn-io/bpmn-properties-panel#178]
__Is your feature request related to a problem? Please describe__

The user raising this issue, uses the *element templates* feature to let users easily configure elements.
Notice that *element templates* allow to configure whether a custom input field is allowed to be empty or not (via the `notEmpty` constraint).

Now in some cases the users of the templated elements, leave custom input fields empty (which does not create an issue in the propPanel UI, since `notEmpty` was not set). However, this then makes in many cases the process not deployable, since the respective injected element, is not configured valid.

Concrete example:
1. Have a field injections template (notice that the 2nd field injection does not have `notEmpty` set)
```
{
    "$schema": "https://unpkg.com/@camunda/element-templates-json-schema@0.4.0/resources/schema.json",
    "name": "Field Injections",
    "id": "com.camunda.example.FieldInjections",
    "appliesTo": [
      "bpmn:ServiceTask"
    ],
    "properties": [
      {
        "label": "Field Injection String",
        "type": "String",
        "value": "My Field Injection Value",
        "binding": {
          "type": "camunda:field",
          "name": "fieldInjectionString",
          "expression": false
        },
      "constraints": {
        "notEmpty": true
      }
      },
      {
        "label": "Field Injection Expression",
        "type": "String",
        "value": "${fieldInjectionExpression}",
        "editable": false,
        "binding": {
          "type": "camunda:field",
          "name": "fieldInjectionExpression",
          "expression": true
        }
      }
    ]
  }
```
2. Apply the template
3. Leave the 2nd custom field empty
4. => You can now not deploy this process, because still two field injections were set in the XML, but the second one is not configured with a valuel.

__Describe the solution you'd like__

In the example case described above, it would be great if the second field injection would not be part of the XML (if the user leaves the respective custom input empty). This could be implemented in two ways:

A) always don't insert elements, if the respective inputs are left blank
B) don't insert elements, if the repspective inputs are left blank and an additional flag was set (e.g., `optional`)

__Describe alternatives you've considered__

See solutions.

__Additional context__

Raised via https://jira.camunda.com/browse/SUPPORT-12176

**Comment by barmac** [comment]
Hmm so this means we would have to implement something similar to the input/output toggle feature where the inputs/outputs toggled off are not represented in the XML.

**Comment by MaxTru** [comment]
> Hmm so this means we would have to implement something similar to the input/output toggle feature where the inputs/outputs toggled off are not represented in the XML.

Yes, probably. So it is good that we have it as prototype (the input/output toggle) :-)

**Comment by nikku** [comment]
I'm highly in favor of introducing an `optional` field in this case. 

If declared as `optional` a property will only be serialized if not empty.

Just changing the existing behavior (not serializing `notEmpty=false` marked properties anymore) is a breaking change that I'd like to avoid.
